rfcs/2020-05-25-2685-dev-workflow-simplification.md

```diff
@@ -0,0 +1,244 @@
+
+It proposes these changes such that our team and contributors will have to explicitly **opt in** to virtualization/containerization/shell-manipulation, and attempts to do it in a convienent, unsurprising way.
+
+## Motivation
+
+Currently, Vector's build system is a wild, powerful beast with a lot of amazing features and mechanisms. Unfortunately, complexity has kind of gotten away from us, and we need to reclaim ownership of the automations.
+
+Notably, our system works right now, as is, and we are not strongly motivated to replace or rebuild things just for the fun of it. We are motivated to make small, forward looking changes that improve the maintainability of our product.
+
+**Before we do this, we should make sure we agree this offers realistic benefits to our future maintenance.**
+
+## Guide-level Proposal
+
+> **Note:** Replaces the **Development → Setup** section of the `[CONTRIBUTING.md](http://contributing.md)` file.
+
+We're super excited to have you interested in working on Vector! Before you start you should pick how you want to develop.
+
+For small or first-time contributions, we recommend the Docker method. If you do a lot of contributing, try adopting the Nix method! It'll be way faster and feel more smooth. Prefer to do it yourself? That's fine too!
+
+### Use a Docker or Podman environment
+
+> **Targets:** You can use this method to produce AARCH64, Arm6/7, as well as x86/64 Linux builds.
+
+Since not everyone has a full working native environment, or can use Nix, we took our Nix environment and stuffed it into a Docker (or Podman) container!
+
+This is ideal for users who want it to "Just work" and just want to start contributing. It's also what we use for our CI, so you know if it breaks we can't do anything else until we fix it. 😉
+
+**Before you go farther, install Docker or Podman through your official package manager, or from the [Docker](https://docs.docker.com/get-docker/) or [Podman](https://podman.io/) sites.**
+
+```bash
+# Optional: Only if you use `podman`
+export container-tool="podman"
+```
+
+By default, `make environment` style tasks will do a `docker pull` from Github's container repository, you can **optionally** build your own environment while you make your morning coffee ☕:
+
+```bash
+# Optional: Only if you want to go make a coffee
+make environment-prepare
+```
+
+Now that you have your coffee, you can run any of the following:
+
+```bash
+# Enter a shell with optimized mounts for interactive processes.
+# Inside here, you can use Vector like you have full toolchain (See below!)
+make environment
+# Add extra cli opts
+make enviroment cli-opts="--publish 3000:2000"
```

Comment:
How do you feel about using `net=host` for environment by default?

Reply:
Breaks windows/mac compat so no.